python/cudf/cudf/api/types/categoricals.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+from typing import List, Union
+
+import cudf
+
+ParentType = Union["cudf.Series", "cudf.Index"]
+
+
+def _union_categoricals(
```

Comment:
Given that an external project (Dask) is using this function, I think we should make it public. Marking it as internal may lead us to believe in the future that we can freely delete or change it.

Reply:
An external project is not using the function, dask-cudf is registering a dispatch for the function so it's entirely within our control. The plan was to make it public when we're ready to align to https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.api.types.union_categoricals.html, but that requires supporting `cudf.Categorical` first, which we're on our way to do anyway.

Reply:
Added this into `TODO`